### PR TITLE
Psd views

### PR DESCRIPTION
**Pull request by AdarshNandanwar**

### CODE REVIEW

**Comment by ChetanKhanna** [comment]
Rebase and resolve conflicts.

File: transfers/views/deadline_views.py [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    initial = {'key': 'value'}
+    template_name = 'transfers/psd_dashboard.html'
+    context = {}
+
+    def get(self, request, *args, **kwargs):
+        form = self.form_class(initial=self.initial)
+        self.context = {
+            'form': form
+        }
+        return render(request, self.template_name, self.context)
```
Comment (ChetanKhanna) [inline transfers/views/deadline_views.py]: `get` method is not returning a `JsonResponse`, is this temporary?

File: transfers/views/deadline_views.py [hunk 2]
```diff
@@ -0,0 +1,31 @@
+from transfers.constants import TransferType
+from django.http import JsonResponse
+from django.shortcuts import render
+from django.views import generic
+ 
+from transfers.models import DeadlineModel
+ 
+from transfers.utils import post_PSD_post
+from transfers.forms import PSDForm
```
Comment (ChetanKhanna) [inline transfers/views/deadline_views.py]: remove unnecessary imports

File: transfers/utils.py [hunk 3]
```diff
@@ -39,6 +40,52 @@ def _get_ts2ps_application_status(userprofile, ts2ps):
...
+def post_PSD_post(form):
+    try:
+        update_psd = DeadlineModel.objects.all().first()
+    except:
+        update_psd = DeadlineModel.objects.create()
+    update_psd.deadline_PS2TS = form.cleaned_data.get('deadline_PS2TS')
+    update_psd.deadline_TS2PS = form.cleaned_data.get('deadline_TS2PS')
+    update_psd.message = form.cleaned_data.get('message')
+    update_psd.is_active_PS2TS = form.cleaned_data.get('is_active_PS2TS')
+    update_psd.is_active_TS2PS = form.cleaned_data.get('is_active_PS2TS')
```
Comment (ChetanKhanna) [inline transfers/utils.py:151]: Leave a line.

File: transfers/utils.py [hunk 4]
```diff
@@ -39,6 +40,52 @@ def _get_ts2ps_application_status(userprofile, ts2ps):
...
+        }
+        response['error'] = False
+        response['message'] = 'success'
+    except Exception as e:
+        response['data'] = {}
+        response['error'] = True
+        response['message'] = 'error'
+    return JsonResponse(response, safe=False)
+
+def post_PSD_post(form):
```
Comment (ChetanKhanna) [inline transfers/utils.py]: won't it be `update_psd_data`?

File: transfers/utils.py [hunk 5]
```diff
@@ -39,6 +40,52 @@ def _get_ts2ps_application_status(userprofile, ts2ps):
...
+            'is_active_TS2PS': is_active_TS2PS,
+            'message': message
+        }
+        response['error'] = False
+        response['message'] = 'success'
+    except Exception as e:
+        response['data'] = {}
+        response['error'] = True
+        response['message'] = 'error'
+    return JsonResponse(response, safe=False)
```
Comment (ChetanKhanna) [inline transfers/utils.py]: This function returns a response, will not come here. Move it somewhere inside `psd_views`

File: transfers/urls.py [hunk 6]
```diff
@@ -2,14 +2,15 @@
...
-    hod_views, supervisor_views, assoc_dean_views)
+    hod_views, supervisor_views, deadline_views, assoc_dean_views)
 
 
 urlpatterns = [
     path('login-redirect/', redirect_views.login_redirect_view),
     path('student/PS2TS/', student_views.PS2TSFormView.as_view()),
     path('student/TS2PS/', student_views.TS2PSFormView.as_view()),
     path('student/dashboard/', student_views.StudentDashboardView.as_view()),
+    path('psd/dashboard/', deadline_views.PSDview.as_view()),
```
Comment (ChetanKhanna) [inline transfers/urls.py]: rename the file to `psd_views` instead of `deadline_views`

**Comment by ChetanKhanna** [comment]
@nrupeshsurya @AdarshNandanwar

File: transfers/views/deadline_views.py (hunk 1 above)
Comment (AdarshNandanwar) [reply transfers/views/deadline_views.py]: No. This will return render. The function for JsonResponse is get_PSD_data

**Review by ChetanKhanna — APPROVED** [review]

**Comment by ChetanKhanna** [comment]
Not merging yet, this branch needs rebasing.

File: transfers/utils.py [hunk 7]
```diff
@@ -1,8 +1,15 @@
...
 
 from .constants import UserType, TransferType
 from .models import PS2TSTransfer, TS2PSTransfer
 
+=======
+from .constants import UserType, TransferType
+from django.http import JsonResponse
+from .models import PS2TSTransfer, TS2PSTransfer, DeadlineModel
+from django.utils import timezone as datetime
+>>>>>>> PSDViews, model changes, some other people's bugs fixed
```
Comment (ChetanKhanna) [inline transfers/utils.py]: conflict markers

File: transfers/urls.py [hunk 8]
```diff
@@ -12,6 +16,13 @@
     path('student/PS2TS/', student_views.PS2TSFormView.as_view()),
     path('student/TS2PS/', student_views.TS2PSFormView.as_view()),
     path('student/dashboard/', student_views.StudentDashboardView.as_view()),
+<<<<<<< HEAD
+    path('psd/dashboard/', deadline_views.PSDview.as_view()),
+=======
+    path('psd/dashboard/', psd_views.PSDview.as_view()),
+    path('hod/home/', hod_views.HODHomeView.as_view()),
+    path('supervisor/home/', supervisor_views.SupervisorHomeView.as_view()),
+>>>>>>> update get_PSD_data and other minor changes
```
Comment (ChetanKhanna) [inline transfers/urls.py]: conflict markers

File: transfers/urls.py [hunk 9]
```diff
@@ -2,7 +2,11 @@
 from django.conf.urls import url
 
 from transfers.views import (redirect_views, student_views,
-    hod_views, supervisor_views, assoc_dean_views)
+<<<<<<< HEAD
+    hod_views, supervisor_views, assoc_dean_views, deadline_views)
+=======
+    hod_views, supervisor_views, psd_views, assoc_dean_views)
+>>>>>>> update get_PSD_data and other minor changes
```
Comment (ChetanKhanna) [inline transfers/urls.py]: Conflict markers.

**Review by ChetanKhanna — COMMENTED** [review]
You seem to have forgotten to actually resolve the conflicts. There are conflict markers everywhere.

**Review by ChetanKhanna — APPROVED** [review]

**Pull request merged**